commit 6b5445891f7c4cf39b41681e51bf3ce25aa728d2 (refs/changes/77/3677/4)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-03-19T19:32:27+00:00 (1 year, 7 months ago)
    
    OS-6742 "vmadm reboot" is not implemented for bhyve
    Reviewed by: Dave Eddy <dave.eddy@joyent.com>
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 5fb5caf7..c16112d1 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -14426,21 +14426,21 @@ function reset(uuid, log, callback)
 }
 
 /*
- * This handles rebooting docker=true zones.
+ * This handles rebooting docker=true and bhyve zones.
  */
-function doDockerReboot(vmobj, options, callback)
+function doStopStartReboot(vmobj, options, callback)
 {
     var log = options.log;
     var tracers_obj;
 
     if (process.env.EXPERIMENTAL_VMJS_TRACING) {
-        tracers_obj = traceUntilCallback('docker-reboot', log, callback);
+        tracers_obj = traceUntilCallback('stopstart-reboot', log, callback);
         callback = tracers_obj.callback;
         log = tracers_obj.log;
     }
 
     log.debug({vmobj_pid: vmobj.pid, timeout: options.timeout},
-        'doDockerReboot');
+        'doStopStartReboot');
 
     if (!options.timeout) {
         options.timeout = 10;
@@ -14677,8 +14677,8 @@ exports.reboot = function (uuid, options, callback)
             // re-add log in case it was changed
             options.log = log;
 
-            if (vmobj.docker) {
-                doDockerReboot(vmobj, options, cb);
+            if (vmobj.docker || vmobj.brand === 'bhyve') {
+                doStopStartReboot(vmobj, options, cb);
             } else {
                 doReboot(vmobj, options, cb);
             }
